AsyncDisplayKit/ASViewController.m

```diff
@@ -0,0 +1,47 @@
+//
+//  ASViewController.m
+//  AsyncDisplayKit
+//
+//  Created by Huy Nguyen on 16/09/15.
+//  Copyright (c) 2015 Facebook. All rights reserved.
+//
+
+#import "ASViewController.h"
+#import "ASAssert.h"
+#import "ASDimension.h"
+
+@implementation ASViewController
+
+- (instancetype)initWithNode:(ASDisplayNode *)node
+{
+  if (!(self = [super init])) {
+    return nil;
+  }
+  
+  ASDisplayNodeAssertNotNil(node, @"Node must not be nil");
+  ASDisplayNodeAssertTrue(!node.layerBacked);
+  _node = node;
+  
+  return self;
+}
+
+- (void)loadView
+{
+  ASDisplayNodeAssertTrue(!_node.layerBacked);
+  self.view = _node.view;
+}
+
+- (void)viewWillLayoutSubviews
+{
+  ASSizeRange constrainedSize = ASSizeRangeMake(self.view.bounds.size, self.view.bounds.size);
+  [_node measureWithSizeRange:constrainedSize];
```

Comment:
Sorry for misunderstood your question. That's indeed a tricky question. Yeah I think the behaviour would be undefined, at least for the current implementation. My main reason is that the async layout pass can finished right before or after any of these events: VC is pushed, loadView, viewWillLayoutSubviews, etc. If we have block on the remainder in, say, viewDidAppear, we may be able to do final set up there in case the background operation missed these events.

(There is a high chance that I'm talking non-sense here, please take it with a grain of salt haha)